lib/Modules/FreeRdpHooks/FreeRdpHooks.psm1

```diff
@@ -24,33 +24,26 @@ Import-Module JujuHooks
 Import-Module OpenStackCommon
 
 function Get-Vcredist {
-    BEGIN {
+    Write-JujuWarning "Getting vcredist."
+    $vcredistUrl = Get-JujuCharmConfig -Scope "vcredist-url"
+    if(!$vcredistUrl) {
         $resourcesSupport = (Get-Command resource-get.exe -ErrorAction SilentlyContinue) -ne $null
```

Comment:
You don't have to retry on all exceptions. We can create a PR for either ```Start-ExecuteWithRetry``` to accept terminal exceptions, or we can do a PR to change ```Get-JujuResource``` to add a ```Start-ExecuteWithRetry```. Either of those 2 options is better then accepting a hack :)